redaxo/src/addons/backup/lib/backup.php

```diff
@@ -362,12 +362,11 @@ public static function exportDb($filename, array $tables = null)
      * Exportiert alle Ordner $folders aus dem Verzeichnis /files.
      *
      * @param array $folders Array von Ordnernamen, die exportiert werden sollen
-     *
-     * @return string Inhalt des Tar-Archives als String
      */
-    public static function exportFiles($folders)
+    public static function exportFiles($folders, $archivePath)
```

Comment:
Hmm neuer param == BC break?

Reply:
Ja stimmt, da wir hier nicht in internal sind. 
warum hattest du hier den param eingeführt?

Reply:
Wir können den aufrufenden code auch hier in die methode verlagern

Reply:
Ahh nee .. vorher haben wir den steing returned und das file geschrieben.. mit dem Pr schreiben wir das file hier, damit wir den file-content nicht komplett im php speicher halten müssen

Reply:
Also siehst du eine Möglichkeit, wie wir das Problem irgendwie lösen?

Reply:
Nicht ohne die api zu verändern.

Einzige idee wg bc: die alte klasse 1:1 drinnen lassen und die neue logik in einer anderen neuen klasse implementieren

Reply:
Scheinbar wird die Methode in einem Release-Skript in einer Demo verwendet: https://github.com/FriendsOfREDAXO/demo_fullpage/blob/3ec3168b551068df71b0603c3d2ee586237f6e8f/release/addon_release.php#L256-L257

Vielleicht, wenn der zweite Param fehlt, dann in temporäre Datei packen und den Inhalt wie bisher returnen?

Reply:
hab das versucht so einzubauen (ungetestet)

Reply:
Der Parameter muss denke ich optional sein mit default-wert, sonst kommt es zu einer warning oder so, wenn man es mit einem parameter aufruft.